sql/catalog/lease: TestDescriptorRefreshOnRetry failed

sql/catalog/lease.TestDescriptorRefreshOnRetry [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15755141?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15755141?buildTab=artifacts#/) on master @ [0003c0966b0daa019a47bc3884f8ad9a6c6d99a0](https://github.com/cockroachdb/cockroach/commits/0003c0966b0daa019a47bc3884f8ad9a6c6d99a0):


```
=== RUN   TestDescriptorRefreshOnRetry
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/5b16c701c1da9e1ccd7b3d919dbf90f6/logTestDescriptorRefreshOnRetry4081217888
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_test.go:996: condition failed to evaluate within 45s: from lease_test.go:998: didnt release descriptor, 2 != 1
    panic.go:626: -- test log scope end --
--- FAIL: TestDescriptorRefreshOnRetry (48.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121416 sql/catalog/lease: TestDescriptorRefreshOnRetry failed [load-job-query error during test server startup] [A-jobs C-test-failure O-robot P-1 T-jobs branch-release-23.1]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDescriptorRefreshOnRetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39692

Jira issue: CRDB-39692

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -948,7 +948,11 @@ func TestDescriptorRefreshOnRetry(t *testing.T) {
 	srv, sqlDB, kvDB := serverutils.StartServer(t, params)
 	defer srv.Stopper().Stop(context.Background())
 	s := srv.ApplicationLayer()
-
+	// Disable the automatic stats collection, which could interfere with
+	// the lease acquisition counts in this test.
+	if _, err := sqlDB.Exec("SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false"); err != nil {
+		t.Fatal(err)
+	}
 	if _, err := sqlDB.Exec(`
 CREATE DATABASE t;
 CREATE TABLE t.foo (v INT);
@@ -1313,6 +1317,11 @@ CREATE DATABASE t;
 CREATE TABLE t.test1 (k CHAR PRIMARY KEY, v CHAR);
 CREATE TABLE t.test2 ();
 `); err != nil {
+		t.Fatal(err)
+	}
+	// Disable the automatic stats collection, which could interfere with
+	// the lease acquisition counts in this test.
+	if _, err := t.db.Exec("SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false"); err != nil {
 		t.Fatal(err)
 	}
 

